Added support for tags on projects

garethr opened the pull request:
I standardized on delete as the verb here, to match the other API
methods. It's tags/delete on group and tags/remove on groups. It's
delete on projects.

I prefer the high-level consistency. We can add an alias if this ends up
confusing folks.

nirfuchs commented on snyk/models.py:
```diff
@@ -430,6 +430,7 @@ class Project(DataClassJSONMixin):
     issueCountsBySeverity: IssueCounts
     imageTag: Optional[str] = None
     imageId: Optional[str] = None
+    tags: Optional[List[Any]] = field(default_factory=list)
```
May I propose we make `tags` a type produced by `Manager.factory`?
if this type would have `all(), add(), delete()` methods then the calls on project would look like:
`project.tags.all()` - list tags for a project
`project.tags.add(**kwargs)` - add tags for project, unfortunately this will require N calls as the tags POST API allows specifying a single key and value attributes
`project.tags.delete(**kwargs)` - delete tags for project, unfortunately this will require N calls to the tags DELETE API for the same reasons above.

garethr commented:
I thought about going the manager route, but didn't to avoid additional requests for loading tags. But it's definitely nicer for the add and delete methods. I've added a simple cache on the project which avoids the extra request for tags and moved everything to a manager.
